src/format.rs

```diff
@@ -0,0 +1,366 @@
+                    }
+                    (None | Some(Padding::Zeros), width) => {
+                        let width = width.unwrap_or(2);
+                        write!(f, "{:0width$}", century)
+                    }
+                }
+            }
+            Spec::YearRem100 => {
+                let year_rem_100 = time.year().rem_euclid(100);
+
+                match (self.padding, self.width) {
+                    (Some(Padding::None), _) => write!(f, "{}", year_rem_100),
+                    (Some(Padding::Spaces), width) => {
+                        let width = width.unwrap_or(2);
+                        write!(f, "{: >width$}", year_rem_100)
+                    }
+                    (None | Some(Padding::Zeros), width) => {
+                        let width = width.unwrap_or(2);
+                        write!(f, "{:0width$}", year_rem_100)
+                    }
+                }
+            }
+            Spec::Month => todo!(),
+            Spec::MonthName => todo!(),
+            Spec::MonthNameAbbr => todo!(),
+            Spec::MonthDayZero => todo!(),
+            Spec::MonthDaySpace => todo!(),
+            Spec::YearDay => todo!(),
+            Spec::Hour24hZero => todo!(),
+            Spec::Hour24hSpace => todo!(),
+            Spec::Hour12hZero => todo!(),
+            Spec::Hour12hSpace => todo!(),
+            Spec::MeridianLower => todo!(),
+            Spec::MeridianUpper => todo!(),
+            Spec::Minute => todo!(),
+            Spec::Second => todo!(),
+            Spec::MilliSecond => todo!(),
+            Spec::FractionalSecond => todo!(),
+            Spec::TimeZoneOffsetHourMinute => todo!(),
+            Spec::TimeZoneOffsetHourMinuteColon => todo!(),
+            Spec::TimeZoneOffsetHourMinuteSecondColon => todo!(),
+            Spec::TimeZoneOffsetHour => todo!(),
+            Spec::TimeZoneName => todo!(),
+            Spec::WeekDayName => todo!(),
+            Spec::WeekDayNameAbbr => todo!(),
+            Spec::WeekDayFrom1 => todo!(),
+            Spec::WeekDayFrom0 => todo!(),
+            Spec::WeekBasedYear => todo!(),
+            Spec::WeekBasedYearRem100 => todo!(),
+            Spec::WeekNumberOfWeekBasedYear => todo!(),
+            Spec::WeekNumberFromSunday => todo!(),
+            Spec::WeekNumberFromMonday => todo!(),
+            Spec::SecondsSinceEpoch => todo!(),
+            Spec::Newline => todo!(),
+            Spec::Tabulation => todo!(),
+            Spec::Percent => todo!(),
+            Spec::CombinationDateTime => todo!(),
+            Spec::CombinationDate => todo!(),
+            Spec::CombinationIso8601 => todo!(),
+            Spec::CombinationVmsDate => todo!(),
+            Spec::CombinationTime12h => todo!(),
+            Spec::CombinationHourMinute24h => todo!(),
+            Spec::CombinationTime24h => todo!(),
+        }
+    }
+}
+
+pub(crate) struct TimeFormatter<'a> {
+    time: &'a Time,
+    format: &'a str,
```

Comment:
I think these should be two parameterized lifetimes, there's no need for the to be exactly the same. This would prevent using e.g. a `&'static str` and a stack-local `Time`.